File: snippets/node/wallet-address/wallet-address-get-keys.ts
import dotenv from "dotenv";
import { join } from "path";
import { fileURLToPath } from "url";

dotenv.config({
    path: fileURLToPath(join(import.meta.url, "..", "..", ".env")),
});

const KEY_ID = process.env.KEY_ID;
const WALLET_ADDRESS = process.env.WALLET_ADDRESS;
const PRIVATE_KEY_PATH = process.env.PRIVATE_KEY_PATH;

//@! start chunk 1 | title=Import dependencies
import { createAuthenticatedClient } from "@interledger/open-payments";
//@! end chunk 1

//@! start chunk 2 | title=Initialize Open Payments client
const client = await createAuthenticatedClient({
    walletAddressUrl: WALLET_ADDRESS,
    privateKey: PRIVATE_KEY_PATH,
    keyId: KEY_ID,
});
//@! end chunk 2

//@! start chunk 3 | title=Get wallet address keys
const walletAddressKeys = await client.walletAddress.getKeys({
    url: WALLET_ADDRESS,
});
//@! end chunk 3

//@! start chunk 4 | title=Output
console.log("WALLET ADDRESS KEYS:", JSON.stringify(walletAddressKeys, null, 2));
//@! end chunk 4

File: snippets/go/wallet-address/wallet-address-get-keys.go
package main

//@! start chunk 1 | title=Import dependencies
import (
	"context"
	"encoding/json"
	"fmt"
	"log"
	"os"

	op "github.com/interledger/open-payments-go"
)

//@! end chunk 1

func main() {
	walletAddressURL := os.Getenv("WALLET_ADDRESS")
	if walletAddressURL == "" {
		log.Fatalf("WALLET_ADDRESS environment variable is required\n")
	}
	privateKeyBase64 := os.Getenv("PRIVATE_KEY")
	if privateKeyBase64 == "" {
		log.Fatalf("PRIVATE_KEY environment variable is required\n")
	}
	keyID := os.Getenv("KEY_ID")
	if keyID == "" {
		log.Fatalf("KEY_ID environment variable is required\n")
	}

	//@! start chunk 2 | title=Initialize Open Payments client
	client, err := op.NewAuthenticatedClient(walletAddressURL, privateKeyBase64, keyID)
	if err != nil {
		log.Fatalf("Error creating authenticated client: %v\n", err)
	}
	//@! end chunk 2

	//@! start chunk 3 | title=Get wallet address keys
	walletAddressKeys, err := client.WalletAddress.GetKeys(context.TODO(), op.WalletAddressGetKeysParams{
		URL: walletAddressURL,
	})
	if err != nil {
		log.Fatalf("Error fetching wallet address keys: %v\n", err)
	}
	//@! end chunk 3

	//@! start chunk 4 | title=Output
	keysJSON, err := json.MarshalIndent(walletAddressKeys, "", "  ")
	if err != nil {
		log.Fatalf("Error marshaling wallet address keys: %v\n", err)
	}
	fmt.Println("WALLET ADDRESS KEYS:", string(keysJSON))
	//@! end chunk 4
}

